apps/arena/lib/arena/game/skill.ex

```diff
@@ -330,22 +317,12 @@ defmodule Arena.Game.Skill do
     Enum.reduce(effect.effect_mechanics, player, fn {mechanic_name, mechanic_params} = mechanic, player ->
       should_re_apply? =
         is_nil(Map.get(mechanic_params, :last_application_time)) or
-          now - Map.get(mechanic_params, :last_application_time) >=
-            mechanic_params.effect_delay_ms
+          (not is_nil(Map.get(mechanic_params, :effect_delay_ms)) and
+             now - Map.get(mechanic_params, :last_application_time) >= mechanic_params.effect_delay_ms)
 
       if should_re_apply? do
         do_effect_mechanics(game_state, player, effect, mechanic)
-        |> put_in(
-          [
-            :aditional_info,
-            :effects,
-            effect.id,
-            :effect_mechanics,
-            mechanic_name,
-            :last_application_time
-          ],
-          now
-        )
+        |> Effect.put_in_effect(effect, [:effect_mechanics, mechanic_name, :last_application_time], now)
```

Comment:
what would happend if we need to apply two effect with the same name at the same time? for example when someone get inside two valtimer's pools, the name will be the same but we should do the damage in two different ocurrences

Reply:
There are no restrictions in the general case, but there is `Effect.put_non_owner_stackable_effect/4` which does check for something like that, it checks owner and effect name, so to answer you it will depend on how we are putting the effects on the player and if we are talking same owner or different owners

Reply:
nice!